File: lib/python/opentoken/src/main/opentoken/io/metadata_writer.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from abc import ABC, abstractmethod
from typing import Dict, Any


class MetadataWriter(ABC):
    """Abstract base class for writing metadata."""

    def __init__(self, output_path: str):
        """
        Initialize the metadata writer.

        Args:
            output_path: The output file path
        """
        self.output_path = output_path

    @abstractmethod
    def write(self, metadata_map: Dict[str, Any]) -> None:
        """
        Write metadata to the output destination.

        Args:
            metadata_map: The metadata to write
        """
        pass

File: lib/java/opentoken/src/main/java/com/truveta/opentoken/io/MetadataWriter.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.io;

import java.io.IOException;
import java.util.Map;

/**
 * Interface for writing metadata to a specified output format.
 * Implementations of this interface should handle the specifics of writing
 * metadata in the desired format, such as JSON, CSV, or Parquet.
 * Each call to write will generate a new metadata file based on the input given.
 */
public interface MetadataWriter {

    /**
     * Writes metadata to the output destination.
     * 
     * @param metadataMap a map containing metadata key-value pairs.
     * @throws IOException if an error occurs while writing the metadata.
     *                     This method should handle the specifics of the output
     *                     format,
     *                     such as JSON, CSV, or Parquet.
     */
    void write(Map<String, Object> metadataMap) throws IOException;
}
